tests/e2e/tools/FFI/bluechi-evil-agent/evil-agent.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/godbus/dbus/v5"
+)
+
+// service, object and register interface for basic client connection
+const (
+	BcDusInterface = "org.eclipse.bluechi"
+	BcObjectPath   = "/org/eclipse/bluechi/internal"
+	MethodRegister = "org.eclipse.bluechi.internal.Manager.Register"
+)
+
+// object path and signals of internal agents interface
+// see https://github.com/containers/bluechi/blob/main/data/org.eclipse.bluechi.internal.Agent.xml#L81-L116
+const (
+	InternalAgentObject = "/org/eclipse/bluechi/internal/agent"
+
+	InternalAgentSignalJobDone = "org.eclipse.bluechi.internal.Agent.JobDone"
+	InternalAgentSignalUnitNew = "org.eclipse.bluechi.internal.Agent.UnitNew"
+	// ...
+)
+
+func main() {
+	if len(os.Args) != 4 {
+		fmt.Printf("Usage: %s <controller_ip> <controller_port> <node_name>", os.Args[0])
+		os.Exit(1)
+	}
+
+	ctrlIP := os.Args[1]
+	ctrlPort := os.Args[2]
+	nodeName := os.Args[3]
+
+	address := fmt.Sprintf("tcp:host=%s,port=%s", ctrlIP, ctrlPort)
```

Comment:
I recommend just taking a dbus-style address on the command-line instead. Then you can support unix sockets, etc. Like we're using with the TLS proxy.